src/webgpu/util/conversion.ts

```diff
@@ -503,14 +503,19 @@ export class Scalar {
       case -Infinity:
         return Colors.bold(this.value.toString());
       default: {
+        // Uint8Array.map returns a Uint8Array, so cannot use .map directly
+        const hex = Array.from(this.bits)
+          .reverse()
+          .map(x => x.toString(16).padStart(2, '0'))
+          .join('');
         const n = this.value as Number;
         if (n !== null) {
           return (
             Colors.bold(this.value.toString()) +
-            `(0x${this.value.toString(16)}, subnormal: ${isSubnormalNumber(n.valueOf())})`
+            ` (hex: 0x${hex}, subnormal: ${isSubnormalNumber(n.valueOf())})`
```

Comment:
I think the "hex: " prefix is redundant here.

Reply:
+1

Also:

* How can you tell the difference between an integer and float without looking at the hex value? `1 (0x00000001)` vs `1 (0x3F800000)`? What do we think about always printing a `.0` if the number has no fractional part?

* I'm a bit concerned about the verbosity of always printing `subnormal: false`. Isn't this used to print the scalars of a vector?
What do we think about only printing `subnormal` if the number... is subnormal? e.g. `1 (0x3F800000)` vs `5.877472e-39 (0x00400000 subnormal)`

Reply:
> What do we think about only printing subnormal if the number... is subnormal?
I added this feature, and my initial implementation did exactly as you suggest. Ryan felt it was less surprising to have it always output something. This doesn't bother me too much either way.

Reply:
I will drop the `hex:`

For Scalars, to tell the difference `1 (0x00000001)` vs `1 (0x3F800000)`, you would have to check the value in this.type. 

I think I am missing something wrt 'always printing `.0`', wouldn't that just cause more confusion? because both integer adn float one would print as 1.0? or is the intent for them to be the same and use the hex to differentiate?

I don't think I properly considered vectors of scalars when reviewing @amaiorano's original subnormal patch. I buy the verbosity argument, so will change it.

Reply:
Sorry, I mistyped. I meant always print a `.0` **if the type is floating-point**. You're right - this would involve looking at `this.type.kind`. That or we add some `isFloatingPoint()` helper on `Scalar`.